core/controllers/reader_test.py

```diff
@@ -135,13 +135,15 @@ def _init_classify_inputs(self, exploration_id):
             exp_services.get_exploration_by_id(exploration_id).states['Home'])
 
     def _is_string_classifier_called(self, answer):
+        sc = classifier_registry.ClassifierRegistry.get_classifier_by_id(
+            feconf.DEFAULT_STRING_CLASSIFIER)
         string_classifier_predict = (
-            lda_string_classifier.LDAStringClassifier.predict)
+            sc.__class__.predict)
```

Comment:
Hmm, actually `lda_string_classifier.LDAStringClassifier` is doing something that we are not supposed to do and this is the reason why we have the registry in first place. So, though it looks icky but that was the only alternative I could think of. Oh, and I'm not sure how this works here (I'm guessing it needs a reference to the method to track how many times it is called, something which is not required in lda_string_classifier_performance_test). So the usecase is entirely different from lda_string_classifier_performance_test.py.